### PR TITLE
cheats: Use global cheat engine

### LINKED ISSUE #6590: Broken Cheats from version Canary 2465+
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Affected Build(s)

Canary 2466

### Description of Issue

the UI changes that went into effect from canary-2466 to date have broken some cheat codes for several games.

all codes for my current games only work upto version 2465

this also affect's the nightly version, but I didn't check when that UI change went into effect with the nightly version but should be around the same release time as 2466

### Expected Behavior

from build 2465 and lower there's a cheats menu visible from the emulation tab, from 2466+ that has been moved into configure current game

### Reproduction Steps

download build 2465 test the codes (currently tested for pokemon gen 6-7)

download any build after 2466+ and you'll see most of the codes stopped working.

[cheats.zip](https://github.com/citra-emu/citra/files/11630794/cheats.zip)

### Log File

lno log file, but provided screenshots of when the changes took effect.

Before Build -2465
image

after build 2466+
image

### System Configuration

CPU:Intel(R) Core(TM) i9-9960X
GPU/Driver: RTX 3080 Ti
RAM: 128GB
OS: Windows 11

### PR DESCRIPTION
**Pull request by GPUCode**
Fixes #6590

- - -
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/6683)

### CODE REVIEW

File: src/core/core.h [hunk 1]
```diff
@@ -372,7 +370,7 @@ class System {
     std::shared_ptr<Frontend::SoftwareKeyboard> registered_swkbd;
 
     /// Cheats manager
-    std::unique_ptr<Cheats::CheatEngine> cheat_engine;
```
Comment (FearlessTobi) [inline src/core/core.h:375]: Can't this stay a unique or a shared pointer?
This is how we do it for most other stuff in Core too

**Comment by FearlessTobi** [comment]
Android CI failure seems to be relevant.

File: src/core/core.h (hunk 1 above)
Comment (GPUCode) [reply src/core/core.h:375]: Some system components need to be initialized on citra startup and I would argue this is one of them. Movie is also a normal member due to this in #6659 . Eventually may also need to convert more subsystems to normal members, something that yuzu has already [done](https://github.com/yuzu-emu/yuzu/blob/04868ab9da2049429808a265dc50696f6300b2b4/src/core/core.cpp#L481)

**Comment by Steveice10** [comment]
Needs rebase and Android CI fix.

**Comment by Steveice10** [comment]
Continuing in https://github.com/citra-emu/citra/pull/7291

**Pull request closed without merging**